tracing/tracing.go

```diff
@@ -6,78 +6,260 @@ package tracing
-func NewFromEnv(serviceName string, options ...jaegercfg.Option) (io.Closer, error) {
-	cfg, err := jaegercfg.FromEnv()
+func NewFromEnv(serviceName string) (io.Closer, error) {
+	cfg, err := parseTracingConfig()
 	if err != nil {
 		return nil, errors.Wrap(err, "could not load jaeger tracer configuration")
 	}
-
-	if cfg.Sampler.SamplingServerURL == "" && cfg.Reporter.LocalAgentHostPort == "" && cfg.Reporter.CollectorEndpoint == "" {
+	if cfg.samplingServerURL == "" && cfg.agentHostPort == "" && cfg.collectorEndpoint == "" {
 		return nil, ErrBlankTraceConfiguration
 	}
+	return cfg.initJaegerTracerProvider(serviceName)
+}
 
-	return installJaeger(serviceName, cfg, options...)
+// parseJaegerTags Parse Jaeger tags from env var JAEGER_TAGS, example of TAGs format: key1=value1,key2=${value2:value3} where value2 is an env var
+// and value3 is the default value, which is optional.
+func parseJaegerTags(sTags string) ([]attribute.KeyValue, error) {
+	pairs := strings.Split(sTags, ",")
+	res := make([]attribute.KeyValue, 0, len(pairs))
+	for _, p := range pairs {
+		k, v, found := strings.Cut(p, "=")
+		if found {
+			k, v := strings.TrimSpace(k), strings.TrimSpace(v)
+			if strings.HasPrefix(v, "${") && strings.HasSuffix(v, "}") {
+				e, d, _ := strings.Cut(v[2:len(v)-1], ":")
+				v = os.Getenv(e)
+				if v == "" && d != "" {
+					v = d
+				}
+			}
+			if v == "" {
+				return nil, errors.Errorf("invalid tag %q, expected key=value", p)
+			}
+			res = append(res, attribute.String(k, v))
+		} else if p != "" {
+			return nil, errors.Errorf("invalid tag %q, expected key=value", p)
+		}
+	}
+	return res, nil
 }
 
-// ExtractTraceID extracts the trace id, if any from the context.
-func ExtractTraceID(ctx context.Context) (string, bool) {
-	sp := opentracing.SpanFromContext(ctx)
-	if sp == nil {
-		return "", false
+type config struct {
+	agentHost         string
+	collectorEndpoint string
+	agentPort         string
+	samplerType       string
+	samplingServerURL string
+	samplerParam      float64
+	jaegerTags        []attribute.KeyValue
+	agentHostPort     string
+}
+
+// parseTracingConfig facilitates initialization that is compatible with Jaeger's InitGlobalTracer method.
+func parseTracingConfig() (config, error) {
+	cfg := config{}
+	var err error
+
+	// Parse reporting agent configuration
+	if e := os.Getenv(envJaegerEndpoint); e != "" {
+		u, err := url.ParseRequestURI(e)
+		if err != nil {
+			return cfg, errors.Wrapf(err, "cannot parse env var %s=%s", envJaegerEndpoint, e)
+		}
+		cfg.collectorEndpoint = u.String()
+	} else {
+		useEnv := false
+		host := envJaegerDefaultUDPSpanServerHost
+		if e := os.Getenv(envJaegerAgentHost); e != "" {
+			host = e
+			useEnv = true
+		}
+
+		port := envJaegerDefaultUDPSpanServerPort
+		if e := os.Getenv(envJaegerAgentPort); e != "" {
+			port = e
+			useEnv = true
+		}
+
+		if useEnv {
+			cfg.agentHost = host
+			cfg.agentPort = port
+			cfg.agentHostPort = fmt.Sprintf("%s:%s", host, port)
```

Comment:
Consider using `net.JoinHostPort` here for IPv6 support, in case a hostname is not used.